src/slidem-slide-base.js

```diff
@@ -81,7 +83,7 @@ export class SlidemSlideBase extends HTMLElement {
 
   #setStep(step) {
     this.querySelector('[step="1"]')?.toggleAttribute?.('past', step > 1);
-    this.#steps.forEach((el, i) => {
+    this.#steps?.forEach((el, i) => {
```

Comment:
See above

Reply:
I don't understand what you are referring to.

Is the `connectedCallback` not always executed? Because if it is, the value of `this.#steps = Array.from(this.querySelectorAll('[reveal]'))` is always an Array.

Is it possible that `#setStep` is executed before the `connectedCallback`?

Reply:
you can remove the optional chain
I did, and got errors, so I put it back

Reply:
That's odd. I can't reproduce it, if I remove the optional in this branch and run locally, everything works fine. Perhaps it's something to do with how callbacks are sequenced when using declarative shadow roots, or it's something to do with how the build tooling in your app sequences it.

It doesn't cause much harm, and if you have an issue, we can change it to have the conditional.